examples/trading/contracts/escrow/sources/shared.move

```diff
@@ -86,19 +99,25 @@ module escrow::shared {
         locked: Locked<U>,
         ctx: &TxContext,
     ): T {
+        let escrowed = dof::remove<EscrowedObjectKey, T>(&mut escrow.id, EscrowedObjectKey {});
+
         let Escrow {
             id,
             sender,
             recipient,
             exchange_key,
-            escrowed,
         } = escrow;
 
         assert!(recipient == tx_context::sender(ctx), EMismatchedSenderRecipient);
         assert!(exchange_key == object::id(&key), EMismatchedExchangeObject);
 
         // Do the actual swap
         transfer::public_transfer(lock::unlock(locked, key), sender);
+
+        event::emit(EscrowSwapped {
+            escrow_id: object::uid_to_inner(&id),
+        });
+
         object::delete(id);
```

Comment:
Just to jog my memory -- this means that this version is building on top of the version of this example from the PR where I adapt it for shared object deletion, right?

Reply:
Yeah, this PR points to your branch which introduces the changes to test scenario too, adding support for shared object deletion.

Reply:
Ah, I hadn't spotted that, yep that makes sense!